lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -984,31 +1033,115 @@ def default_ami_for_region(region, criteria = {})
 
     def create_winrm_transport(machine_spec, machine_options, instance)
       remote_host = determine_remote_host(machine_spec, instance)
+      username = machine_options[:winrm_username] || 'Administrator'
+      # default to http for now, should upgrade to https when knife support self-signed
+      transport_type = machine_options[:winrm_transport] || 'http'
+      type = case transport_type
+             when 'http'
+               :plaintext
+             when 'https'
+               :ssl
+             end
+      if machine_spec.reference[:winrm_port]
+        port = machine_spec.reference[:winrm_port]
+      else #default port
+        port = case transport_type
+               when 'http'
+                 '5985'
+               when 'https'
+                 '5986'
+               end
+      end
+      endpoint = "#{transport_type}://#{remote_host}:#{port}/wsman"
 
-      port = machine_spec.reference['winrm_port'] || 5985
-      endpoint = "http://#{remote_host}:#{port}/wsman"
-      type = :plaintext
       pem_bytes = get_private_key(instance.key_name)
 
-      # TODO plaintext password = bad
-      password = machine_spec.reference['winrm_password']
-      if password.nil? || password.empty?
-        encrypted_admin_password = instance.password_data.password_data
-        if encrypted_admin_password.nil? || encrypted_admin_password.empty?
-          raise "You did not specify winrm_password in the machine options and no encrytpted password could be fetched from the instance"
+      if machine_options[:winrm_password]
+        password = machine_options[:winrm_password]
+      else # pull from ec2 and store in reference
+        if machine_spec.reference['winrm_encrypted_password']
+          decoded = Base64.decode64(machine_spec.reference['winrm_encrypted_password'])
+        else
+          encrypted_admin_password = instance.password_data.password_data
+          if encrypted_admin_password.nil? || encrypted_admin_password.empty?
+            raise "You did not specify winrm_password in the machine options and no encrytpted password could be fetched from the instance"
+          end
+
+          machine_spec.reference['winrm_encrypted_password']||=encrypted_admin_password
+          # ^^ saves encrypted password to the machine_spec
+          decoded = Base64.decode64(encrypted_admin_password)
         end
-        decoded = Base64.decode64(encrypted_admin_password)
-        private_key = OpenSSL::PKey::RSA.new(pem_bytes)
+        # decrypt so we can utilize
+        private_key = OpenSSL::PKey::RSA.new(get_private_key(instance.key_name))
         password = private_key.private_decrypt decoded
       end
 
+      disable_sspi =  machine_options[:winrm_disable_sspi] || false # default to Negotiate
```

Comment:
Currently, we don't overwrite the transport options once they have been set the first time.  For example you could create a `machine 'my_machine'` and set the `ssl_username` to 'foo' and that is the username we will use for all later occurances of `machine 'my_machine'`.

So for now, lets not try and change that.  If you initially have your machine_options setup to use SSL with thumbprint auth, then thats how it will stay.